### ISSUE TITLE
Bug: blis observe does not set min_tokens per-request, causing observed output token distribution to diverge from workload spec

### ISSUE THREAD

**Issue by sriumcp** [inference-sim/inference-sim#1277]
## Describe the bug

`blis observe` sends each request with `max_tokens = req.MaxOutputLen` (correctly sampled per-request from the workload spec distribution), but sends a **single global** `min_tokens` value (from `--min-tokens N`, default 0) for every request.

Because the real server generates tokens until its own EOS signal fires — which can happen well before `max_tokens` — the actual output length for each request is:

```
actual_output_tokens = min(EOS_position, max_tokens)
```

This means the observed output token distribution is a **left-truncated, compressed version** of the workload spec distribution, not a faithful reproduction of it. Requests where the model naturally generates short responses will produce far fewer tokens than the workload spec intended, silently biasing every downstream calibration metric (E2E latency, ITL, TTFT).

`blis run` does not have this problem: the simulator generates exactly `len(req.OutputTokens)` tokens per request (see `simulator.go:797`), so the distribution is reproduced exactly.

## The fix (in plain terms)

When building each HTTP request body in `requestToPending` (`cmd/observe_cmd.go:824`), set:

```
min_tokens = req.MaxOutputLen   (per-request, not a global constant)
max_tokens = req.MaxOutputLen   (already done)
```

Setting `min_tokens == max_tokens` forces vLLM to suppress EOS tokens until exactly `MaxOutputLen` tokens have been generated, then stop. This makes every observed request produce exactly the number of output tokens the workload spec sampled — matching the simulator's behavior.

## Proof that vLLM supports per-request min_tokens

**1. Both OpenAI-compatible endpoints accept `min_tokens` as a per-request field:**

`vllm/entrypoints/openai/completion/protocol.py:74`:
```python
min_tokens: int = 0
```

`vllm/entrypoints/openai/chat_completion/protocol.py:205`:
```python
min_tokens: int = 0
```

**2. vLLM's SamplingParams validates it per-request and enforces the contract:**

`vllm/sampling_params.py:179-181`:
```python
min_tokens: int = 0
"""Minimum number of tokens to generate per output sequence before EOS or
`stop_token_ids` can be generated"""
```

`vllm/sampling_params.py:430-438`:
```python
if self.min_tokens < 0:
    raise ValueError(...)
if self.max_tokens is not None and self.min_tokens > self.max_tokens:
    raise ValueError(
        f"min_tokens must be less than or equal to "
        f"max_tokens={self.max_tokens}, got {self.min_tokens}."
    )
```

The key constraint: `min_tokens <= max_tokens`. Since we will set both to `req.MaxOutputLen`, this is always satisfied.

**3. Enforcement is done via a built-in logits processor that sets EOS logits to -inf:**

`vllm/v1/sample/logits_processor/builtin.py:233-237`:
```python
def apply(self, logits: torch.Tensor) -> torch.Tensor:
    if self.min_toks:
        # Inhibit EOS token for requests which have not reached min length
        logits.index_put_(self.logits_slice, self.neg_inf_tensor)
    return logits
```

This processor is a **built-in** (always active, not opt-in) and applies independently per request in the batch.

## Where to make the change

**File:** `cmd/observe_cmd.go`

**Current code at line 824:**
```go
func requestToPending(req *sim.Request, reqIndex int, noStreaming, unconstrained bool, minTokens int, prefixes map[string]string, prefixLengths map[string]int, tokensPerWord float64) *PendingRequest {
    ...
    return &PendingRequest{
        ...
        MaxOutputTokens: req.MaxOutputLen,
        ...
        MinTokens:       minTokens,   // global constant for ALL requests
        ...
    }
}
```

**Fix:** Remove the `minTokens int` parameter and derive `MinTokens` directly from `req.MaxOutputLen`:
```go
func requestToPending(req *sim.Request, reqIndex int, noStreaming, unconstrained bool, prefixes map[string]string, prefixLengths map[string]int, tokensPerWord float64) *PendingRequest {
    ...
    return &PendingRequest{
        ...
        MaxOutputTokens: req.MaxOutputLen,
        ...
        MinTokens:       req.MaxOutputLen,  // per-request, always == max_tokens
        ...
    }
}
```

Update the call site at `observe_cmd.go:607` to remove the `minTokens` argument.

The `clampRequestsToMinTokens` call at `observe_cmd.go:386-390` is no longer needed (since `min_tokens == max_tokens` per-request, the vLLM constraint `min_tokens <= max_tokens` is trivially always satisfied).

The `--min-tokens` CLI flag at `observe_cmd.go:148` is superseded. Consider removing it or keeping it as an explicit opt-out (e.g. `--min-tokens 0` to disable exact-length enforcement).

## Exception: unconstrained output mode

When `--unconstrained-output` is set, `max_tokens` is not sent to the server. In this mode, `min_tokens` should stay 0 — the whole point is to let the server decide output length freely. The fix must skip setting `MinTokens` when `unconstrained == true`.

## Known limitation: speculative decoding

vLLM rejects `min_tokens > 1` when speculative decoding is enabled (raises `ValueError` at `vllm/v1/engine/input_processor.py:168-173`). If the observed server uses spec decoding, requests will fail. The existing `warnOnFinishReason` detection in `cmd/observe.go:265` already catches silent `min_tokens` non-compliance. No new detection is needed, but a clearer error message mentioning spec decoding as a likely cause would help.

## CLI command to reproduce

```bash
# With a workload whose output token distribution has mean=512:
./blis observe --server-url http://localhost:8000 --model my-model \
  --workload chatbot --rate 5 --num-requests 100 \
  --trace-header out.yaml --trace-data out.csv

# Inspect out.csv: actual output tokens will cluster far below the workload
# spec mean because the server stops at EOS, not at max_tokens.
```

## Expected behavior

The output token distribution in the trace matches the workload spec distribution, exactly as `blis run` reproduces it.

## Actual behavior

The output token distribution is compressed toward shorter values, silently biasing all downstream calibration metrics (E2E latency, TTFT, ITL).

## Which invariant is violated?

- [x] None / unknown

(Observe-pipeline fidelity gap, not a DES invariant violation.)

## Additional context

Discovered via code review. The root cause is that `requestToPending` (`cmd/observe_cmd.go:824`) accepts a single global `minTokens int` argument rather than deriving it from `req.MaxOutputLen`. The `--min-tokens` flag was added as a manual workaround, but a global constant cannot reproduce a distribution — only per-request `min_tokens == max_tokens` can.
